backend/dataall/tasks/stacks_updater.py

```diff
@@ -18,32 +19,39 @@
 def update_stacks(engine, envname):
     with engine.scoped_session() as session:
 
-        all_datasets: [models.Dataset] = db.api.Dataset.list_all_active_datasets(
-            session
-        )
-        all_environments: [
-            models.Environment
-        ] = db.api.Environment.list_all_active_environments(session)
-        log.info(f'Found {len(all_environments)} environments')
+        all_datasets: [models.Dataset] = db.api.Dataset.list_all_active_datasets(session)
+        all_environments: [models.Environment] = db.api.Environment.list_all_active_environments(session)
+
+        log.info(f'Found {len(all_environments)} environments, triggering update stack tasks...')
         environment: models.Environment
         for environment in all_environments:
-            update_stack(session, envname, environment.environmentUri)
+            update_stack(session=session, envname=envname, target_uri=environment.environmentUri, wait=True)
 
         log.info(f'Found {len(all_datasets)} datasets')
         dataset: models.Dataset
         for dataset in all_datasets:
-            update_stack(session, envname, dataset.datasetUri)
+            update_stack(session=session, envname=envname, target_uri=dataset.datasetUri, wait=False)
 
         return all_environments, all_datasets
 
 
-def update_stack(session, envname, target_uri):
+def update_stack(session, envname, target_uri, wait=False):
     stack: models.Stack = db.api.Stack.get_stack_by_target_uri(
         session, target_uri=target_uri
     )
     cluster_name = Parameter().get_parameter(env=envname, path='ecs/cluster/name')
-    if not Ecs.is_task_running(cluster_name, f'awsworker-{stack.stackUri}'):
-        stack.EcsTaskArn = Ecs.run_cdkproxy_task(stack.stackUri)
+    if not Ecs.is_task_running(cluster_name=cluster_name, started_by=f'awsworker-{stack.stackUri}'):
+        stack.EcsTaskArn = Ecs.run_cdkproxy_task(stack_uri=stack.stackUri)
+        if wait:
+            retries = 1
+            while Ecs.is_task_running(cluster_name=cluster_name, started_by=f'awsworker-{stack.stackUri}'):
+                log.info(f"Update for {stack.name}//{stack.stackUri} is not complete, waiting for 30 seconds...")
+                time.sleep(30)
```

Comment:
Let's first move it to constants. I think that what we discussed around putting an SQS queue in front of ECS-cdkproxy is the way to go instead of this custom queueing

Reply:
Fine to me!